lib/Console/Command/Parse.php

```diff
@@ -23,37 +23,93 @@
 
 use \Symfony\Component\Console\Input\InputInterface;
 use \Symfony\Component\Console\Output\OutputInterface;
+use \Mesamatrix\Parser\OglVersion;
+use \Mesamatrix\Parser\UrlCache;
+use \Mesamatrix\Parser\Hints;
 
 class Parse extends \Symfony\Component\Console\Command\Command
 {
-    protected function configure()
-    {
+    protected $urlCache;
+    protected $statuses;
+
+    protected function configure() {
         $this->setName('parse')
              ->setDescription('Parse data and generate XML')
         ;
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output)
-    {
-        //$gitLog = exec_git("log --pretty=format:%H --reverse " .
-        //    Mesamatrix::$config->getValue("git", "oldest_commit").".. -- ".Mesamatrix::$config->getValue("git", "gl3"), $log);
-        //$initialCommit = rtrim(fgets($log));
-        $initialCommit = "master";
-
-        $cat = new \Mesamatrix\Git\ProcessBuilder(array(
-          'show', $initialCommit.':'.\Mesamatrix::$config->getValue('git', 'gl3')
+    protected function execute(InputInterface $input, OutputInterface $output) {
+		$gl3Path = \Mesamatrix::$config->getValue('git', 'gl3', 'docs/GL3.txt');
+        $gitLog = new \Mesamatrix\Git\ProcessBuilder(array(
+            'log', '--pretty=format:%H|%at|%aN|%cN', '--reverse',
+            \Mesamatrix::$config->getValue('git', 'oldest_commit').'..', '--',
+            $gl3Path
         ));
-        $proc = $cat->getProcess();
-        $this->getHelper('process')->mustRun($output, $proc);
-
-        $hints = new \Mesamatrix\Parser\Hints();
-        $parser = new \Mesamatrix\Parser\OglParser($hints);
-        $matrix = $parser->parse_content($proc->getOutput());
+        $gitLogProc = $gitLog->getProcess();
+        $this->getHelper('process')->mustRun($output, $gitLogProc);
+
+        $commitLines = explode(PHP_EOL, $gitLogProc->getOutput());
+        $hints = new Hints();
+        $matrix = new \Mesamatrix\Parser\OglMatrix();
+        foreach ($commitLines as $commitLine) {
+            list($commitHash, $time, $author, $committer) = explode('|', $commitLine);
+            $commit = new \Mesamatrix\Git\Commit($commitHash, $time, $author, $committer);
+            \Mesamatrix::$logger->info('Parsing GL3.txt for commit '.$commit->getHash());
+            $cat = new \Mesamatrix\Git\ProcessBuilder(array(
+              'show', $commit->getHash().':'.$gl3Path
+            ));
+            $proc = $cat->getProcess();
+            $this->getHelper('process')->mustRun($output, $proc);
+
+            $parser = new \Mesamatrix\Parser\OglParser($hints, $matrix, $commit);
+            $parser->parse_content($proc->getOutput());
```

Comment:
I think it makes sense for `parse*()` to return the matrix - it's not useful here, but the return from a 'parse' operation should be the result of that operation, or in this case the matrix. However, I like the way the matrix is passed into the constructor - one Parser, one Matrix.

I changed the way the commit object is passed around - it gets passed into `parse*()`.